backend/service/audit/audit.go

```diff
@@ -137,29 +139,44 @@ func (c *client) ReadEvents(ctx context.Context, start time.Time, end *time.Time
 	return c.storage.ReadEvents(ctx, start, end)
 }
 
-// This should be called via `go` in order to avoid blocking main exectuion.
-func (c *client) poll() {
-	readAndFanout := func(ctx context.Context) {
-		// TODO(maybe): Backpressure on continued failure.
-		events, err := c.storage.UnsentEvents(ctx)
-		if err != nil {
-			return
-		}
+func (c *client) readAndFanout(ctx context.Context) {
+	// TODO(maybe): Backpressure on continued failure.
+
+	c.sinkWriterScope.Counter("events_fetch").Inc(1)
+	events, err := c.storage.UnsentEvents(ctx)
+	if err != nil {
+		return
+	}
 
-		for _, event := range events {
-			for _, s := range c.sinks {
-				if err := s.Write(event); err != nil {
-					c.logger.Error(
-						"error writing audit event to sink",
-						zap.String("sink", s.name),
-						log.ProtoField("event", event),
-						zap.Error(err),
-					)
-				}
+	numEvents := len(events)
+	c.sinkWriterScope.Gauge("events_received").Update(float64(numEvents))
+	c.sinkWriterScope.Counter("events_total").Inc(int64(numEvents))
+
+	flushTimer := c.sinkWriterScope.Timer("unsent_events_write_all").Start()
+	for _, event := range events {
+		lag := time.Since(event.OccurredAt.AsTime())
+		c.sinkWriterScope.Timer("event_lag").Record(lag)
+
+		for _, s := range c.sinks {
+			c.sinkWriterScope.Counter("event_write.attempt")
```

Comment:
do you think that this is superfluous? We should be able to compute this with success and fail metrics, however this is easier and less computation is required.

Reply:
i'll drop it for now. i have seen alignment issues with counters where they don't add up correctly due to flushing on the edge of a window but that's such a rare case.